pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogicCCL_read_committed failed

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogicCCL_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18437193?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18437193?buildTab=artifacts#/) on master @ [890ebb79b22682f0c2af248e94b4b58bd417f4c5](https://github.com/cockroachdb/cockroach/commits/890ebb79b22682f0c2af248e94b4b58bd417f4c5):


```
=== RUN   TestTenantLogicCCL_read_committed
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/b5b01cdf18c474581f3b037c806fc673/logTestTenantLogicCCL_read_committed2281161177
    test_log_scope.go:76: use -show-logs to present logs inline
[16:16:24] setting distsql_workmem='63468B';
[16:16:24] setting distsql_workmem='63468B';
    logic.go:4403: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b5b01cdf18c474581f3b037c806fc673/logTestTenantLogicCCL_read_committed2281161177
--- FAIL: TestTenantLogicCCL_read_committed (11.27s)
=== RUN   TestTenantLogicCCL_read_committed/end#02
[16:16:33] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/21104/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/read_committed: 107 statements
    logic.go:2273: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/21104/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/read_committed:533: 
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=4537bb95 key=/Tenant/10/Table/117/1 iso=ReadCommitted pri=0.01071747 epo=0 ts=1736439388.425907507,0 min=1736439388.425852525,0 seq=0} lock=true stat=PENDING rts=1736439388.425907507,0 wto=false gul=1736439388.925852525,0
[16:16:33] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/21104/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/read_committed with config 3node-tenant: 107 tests, 1 failures
[16:16:34] --- total progress: 107 statements
--- total: 107 tests, 1 failures
    --- FAIL: TestTenantLogicCCL_read_committed/end#02 (5.46s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogicCCL_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46330

--- a/pkg/ccl/logictestccl/testdata/logic_test/read_committed
+++ b/pkg/ccl/logictestccl/testdata/logic_test/read_committed
@@ -483,10 +483,20 @@ SET SESSION CHARACTERISTICS AS TRANSACTION ISOLATION LEVEL SERIALIZABLE
 
 statement ok
 CREATE USER testuser2 WITH VIEWACTIVITY;
+
+statement ok
 DROP TABLE IF EXISTS t;
+
+statement ok
 CREATE TABLE t(a INT PRIMARY KEY);
+
+statement ok
 INSERT INTO t VALUES(1), (2);
+
+statement ok
 GRANT ALL ON t TO testuser;
+
+statement ok
 GRANT ALL ON t TO testuser2;
 
 user testuser

